From 0d6ca91dd32c457f009adfac7cfb28bb890fbe42 Mon Sep 17 00:00:00 2001
From: Neil Cheng <Neil.Cheng@quantatw.com>
Date: Mon, 8 Jun 2026 17:07:11 +0800
Subject: [PATCH] configurations: rainier: Add FSC configurations

Add initial Rainier fan speed control configurations provided by the
thermal team.

This is the first version and may be refined based on platform
validation results.

Upstream-Status: Submitted [https://gerrit.openbmc.org/c/openbmc/entity-manager/+/91974]

Change-Id: I0464259aff42e9c96edd9e05234f532071bd14fb
Signed-off-by: Neil Cheng <Neil.Cheng@quantatw.com>
---
 .../meta/rainier/rainier_scm_2600.json        | 411 ++++++++++++++++++
 1 file changed, 411 insertions(+)

diff --git a/configurations/meta/rainier/rainier_scm_2600.json b/configurations/meta/rainier/rainier_scm_2600.json
index b5754e2..fb4e946 100644
--- a/configurations/meta/rainier/rainier_scm_2600.json
+++ b/configurations/meta/rainier/rainier_scm_2600.json
@@ -232,6 +232,417 @@
             "Name": "Rainier_SCM_CPLD",
             "Type": "LatticeLCMXO3D_9400Firmware"
         },
+        {
+            "CheckHysteresisWithSetpoint": true,
+            "Class": "temp",
+            "DCoefficient": 0.0,
+            "FFGainCoefficient": 0.0,
+            "FFOffCoefficient": 0.0,
+            "ICoefficient": -0.035,
+            "ILimitMax": 100,
+            "ILimitMin": -50,
+            "Inputs": [
+                "MB_CPU_TEMP_C"
+            ],
+            "Name": "PID_MB_CPU",
+            "NegativeHysteresis": 2.0,
+            "OutLimitMax": 100,
+            "OutLimitMin": 0,
+            "PCoefficient": -5.0,
+            "PositiveHysteresis": 0.0,
+            "SetPoint": 76.0,
+            "SlewNeg": 0.0,
+            "SlewPos": 0.0,
+            "Type": "Pid",
+            "Zones": [
+                "Zone 1"
+            ]
+        },
+        {
+            "CheckHysteresisWithSetpoint": true,
+            "Class": "temp",
+            "DCoefficient": 0.0,
+            "FFGainCoefficient": 0.0,
+            "FFOffCoefficient": 0.0,
+            "ICoefficient": -0.035,
+            "ILimitMax": 100,
+            "ILimitMin": -50,
+            "Inputs": [
+                "MB_DIMM_0_0_TEMP_C",
+                "MB_DIMM_0_1_TEMP_C",
+                "MB_DIMM_0_2_TEMP_C",
+                "MB_DIMM_0_3_TEMP_C",
+                "MB_DIMM_0_4_TEMP_C",
+                "MB_DIMM_0_5_TEMP_C",
+                "MB_DIMM_1_0_TEMP_C",
+                "MB_DIMM_1_1_TEMP_C",
+                "MB_DIMM_1_2_TEMP_C",
+                "MB_DIMM_1_3_TEMP_C",
+                "MB_DIMM_1_4_TEMP_C",
+                "MB_DIMM_1_5_TEMP_C"
+            ],
+            "Name": "PID_MB_DIMM",
+            "NegativeHysteresis": 2.0,
+            "OutLimitMax": 100,
+            "OutLimitMin": 0,
+            "PCoefficient": -3.0,
+            "PositiveHysteresis": 0.0,
+            "SetPoint": 79.0,
+            "SlewNeg": 0.0,
+            "SlewPos": 0.0,
+            "Type": "Pid",
+            "Zones": [
+                "Zone 1"
+            ]
+        },
+        {
+            "CheckHysteresisWithSetpoint": true,
+            "Class": "temp",
+            "DCoefficient": 0.0,
+            "FFGainCoefficient": 0.0,
+            "FFOffCoefficient": 0.0,
+            "ICoefficient": -0.035,
+            "ILimitMax": 100,
+            "ILimitMin": -50,
+            "Inputs": [
+                "MB_E1S0_TEMP_C",
+                "MB_E1S1_TEMP_C"
+            ],
+            "MissingIsAcceptable": [
+                "MB_E1S0_TEMP_C",
+                "MB_E1S1_TEMP_C"
+            ],
+            "Name": "PID_MB_E1S",
+            "NegativeHysteresis": 2.0,
+            "OutLimitMax": 100,
+            "OutLimitMin": 0,
+            "PCoefficient": -3.0,
+            "PositiveHysteresis": 0.0,
+            "SetPoint": 65.0,
+            "SlewNeg": 0.0,
+            "SlewPos": 0.0,
+            "Type": "Pid",
+            "Zones": [
+                "Zone 1"
+            ]
+        },
+        {
+            "CheckHysteresisWithSetpoint": true,
+            "Class": "temp",
+            "DCoefficient": 0.0,
+            "FFGainCoefficient": 0.0,
+            "FFOffCoefficient": 0.0,
+            "ICoefficient": -0.035,
+            "ILimitMax": 100,
+            "ILimitMin": -50,
+            "Inputs": [
+                "MB_VR_CPU_CPU0_TEMP_C",
+                "MB_VR_CPU_MEM0_TEMP_C",
+                "MB_VR_CPU_SYS0_TEMP_C",
+                "MB_VR_CPU_SOC0_TEMP_C",
+                "MB_VR_CPU_CPU1_TEMP_C",
+                "MB_VR_CPU_MEM1_TEMP_C",
+                "MB_VR_CPU_SYS1_TEMP_C",
+                "MB_VR_CPU_SOC0_TEMP_C"
+            ],
+            "Name": "PID_MB_VR",
+            "NegativeHysteresis": 2.0,
+            "OutLimitMax": 100,
+            "OutLimitMin": 0,
+            "PCoefficient": -3.0,
+            "PositiveHysteresis": 0.0,
+            "SetPoint": 100.0,
+            "SlewNeg": 0.0,
+            "SlewPos": 0.0,
+            "Type": "Pid",
+            "Zones": [
+                "Zone 1"
+            ]
+        },
+        {
+            "CheckHysteresisWithSetpoint": true,
+            "Class": "temp",
+            "DCoefficient": 0.0,
+            "FFGainCoefficient": 0.0,
+            "FFOffCoefficient": 0.0,
+            "ICoefficient": -0.035,
+            "ILimitMax": 100,
+            "ILimitMin": -50,
+            "Inputs": [
+                "PADDLE_HSC_TEMP_C"
+            ],
+            "Name": "PID_Paddle_HSC",
+            "NegativeHysteresis": 2.0,
+            "OutLimitMax": 100,
+            "OutLimitMin": 0,
+            "PCoefficient": -3.0,
+            "PositiveHysteresis": 0.0,
+            "SetPoint": 100.0,
+            "SlewNeg": 0.0,
+            "SlewPos": 0.0,
+            "Type": "Pid",
+            "Zones": [
+                "Zone 1"
+            ]
+        },
+        {
+            "CheckHysteresisWithSetpoint": true,
+            "Class": "temp",
+            "DCoefficient": 0.0,
+            "FFGainCoefficient": 0.0,
+            "FFOffCoefficient": 0.0,
+            "ICoefficient": -0.035,
+            "ILimitMax": 100,
+            "ILimitMin": -50,
+            "Inputs": [
+                "NIC_Temp_4"
+            ],
+            "Name": "PID_NIC",
+            "NegativeHysteresis": 2.0,
+            "OutLimitMax": 100,
+            "OutLimitMin": 0,
+            "PCoefficient": -3.0,
+            "PositiveHysteresis": 0.0,
+            "SetPoint": 95.0,
+            "SlewNeg": 0.0,
+            "SlewPos": 0.0,
+            "Type": "Pid",
+            "Zones": [
+                "Zone 1"
+            ]
+        },
+        {
+            "AccumulateSetPoint": true,
+            "CycleIntervalTimeMS": 1000,
+            "FailSafePercent": 100.0,
+            "MinThermalOutput": 0.0,
+            "Name": "Zone 1",
+            "Type": "Pid.Zone",
+            "UpdateThermalsTimeMS": 1000,
+            "ZoneIndex": 1
+        },
+        {
+            "Class": "temp",
+            "Inputs": [
+                "SCM_VIRTUAL_CALIBRATED_INLET_TEMP_C"
+            ],
+            "Name": "Stepwise_SCM_INLET",
+            "NegativeHysteresis": 2.0,
+            "Output": [
+                22.0,
+                28.0,
+                34.0,
+                40.0,
+                95.0
+            ],
+            "PositiveHysteresis": 0.0,
+            "Reading": [
+                25.0,
+                30.0,
+                35.0,
+                40.0,
+                45.0
+            ],
+            "Type": "Stepwise",
+            "Zones": [
+                "Zone 1"
+            ]
+        },
+        {
+            "Class": "temp",
+            "Inputs": [
+                "MB_CPU_TEMP_C"
+            ],
+            "Name": "Stepwise_MB_CPU",
+            "NegativeHysteresis": 3.0,
+            "Output": [
+                22.0,
+                34.0,
+                38.0,
+                42.0,
+                46.0,
+                50.0,
+                95.0
+            ],
+            "PositiveHysteresis": 0.0,
+            "Reading": [
+                65.0,
+                67.0,
+                69.0,
+                71.0,
+                73.0,
+                75.0,
+                82.0
+            ],
+            "Type": "Stepwise",
+            "Zones": [
+                "Zone 1"
+            ]
+        },
+        {
+            "Class": "temp",
+            "Inputs": [
+                "MB_DIMM_0_0_TEMP_C",
+                "MB_DIMM_0_1_TEMP_C",
+                "MB_DIMM_0_2_TEMP_C",
+                "MB_DIMM_0_3_TEMP_C",
+                "MB_DIMM_0_4_TEMP_C",
+                "MB_DIMM_0_5_TEMP_C",
+                "MB_DIMM_1_0_TEMP_C",
+                "MB_DIMM_1_1_TEMP_C",
+                "MB_DIMM_1_2_TEMP_C",
+                "MB_DIMM_1_3_TEMP_C",
+                "MB_DIMM_1_4_TEMP_C",
+                "MB_DIMM_1_5_TEMP_C"
+            ],
+            "Name": "Stepwise_MB_DIMM",
+            "NegativeHysteresis": 2.0,
+            "Output": [
+                22.0,
+                30.0,
+                40.0,
+                45.0,
+                95.0
+            ],
+            "PositiveHysteresis": 0.0,
+            "Reading": [
+                56.0,
+                58.0,
+                65.0,
+                70.0,
+                82.0
+            ],
+            "Type": "Stepwise",
+            "Zones": [
+                "Zone 1"
+            ]
+        },
+        {
+            "Class": "temp",
+            "Inputs": [
+                "MB_E1S0_TEMP_C",
+                "MB_E1S1_TEMP_C"
+            ],
+            "MissingIsAcceptable": [
+                "MB_E1S0_TEMP_C",
+                "MB_E1S1_TEMP_C"
+            ],
+            "Name": "Stepwise_MB_E1S",
+            "NegativeHysteresis": 2.0,
+            "Output": [
+                22.0,
+                30.0,
+                95.0
+            ],
+            "PositiveHysteresis": 0.0,
+            "Reading": [
+                50.0,
+                55.0,
+                68.0
+            ],
+            "Type": "Stepwise",
+            "Zones": [
+                "Zone 1"
+            ]
+        },
+        {
+            "Class": "temp",
+            "Inputs": [
+                "MB_VR_CPU_CPU0_TEMP_C",
+                "MB_VR_CPU_MEM0_TEMP_C",
+                "MB_VR_CPU_SYS0_TEMP_C",
+                "MB_VR_CPU_SOC0_TEMP_C",
+                "MB_VR_CPU_CPU1_TEMP_C",
+                "MB_VR_CPU_MEM1_TEMP_C",
+                "MB_VR_CPU_SYS1_TEMP_C",
+                "MB_VR_CPU_SOC0_TEMP_C"
+            ],
+            "Name": "Stepwise_MB_VR",
+            "NegativeHysteresis": 2.0,
+            "Output": [
+                22.0,
+                30.0,
+                95.0
+            ],
+            "PositiveHysteresis": 0.0,
+            "Reading": [
+                80.0,
+                90.0,
+                115.0
+            ],
+            "Type": "Stepwise",
+            "Zones": [
+                "Zone 1"
+            ]
+        },
+        {
+            "Class": "temp",
+            "Inputs": [
+                "PADDLE_HSC_TEMP_C"
+            ],
+            "Name": "Stepwise_Paddle_HSC",
+            "NegativeHysteresis": 2.0,
+            "Output": [
+                22.0,
+                30.0,
+                95.0
+            ],
+            "PositiveHysteresis": 0.0,
+            "Reading": [
+                80.0,
+                90.0,
+                115.0
+            ],
+            "Type": "Stepwise",
+            "Zones": [
+                "Zone 1"
+            ]
+        },
+        {
+            "Class": "temp",
+            "Inputs": [
+                "NIC_Temp_4"
+            ],
+            "Name": "Stepwise_NIC",
+            "NegativeHysteresis": 2.0,
+            "Output": [
+                22.0,
+                30.0,
+                95.0
+            ],
+            "PositiveHysteresis": 0.0,
+            "Reading": [
+                80.0,
+                90.0,
+                100.0
+            ],
+            "Type": "Stepwise",
+            "Zones": [
+                "Zone 1"
+            ]
+        },
+        {
+            "Class": "temp",
+            "Inputs": [
+                "MB_INLET_TEMP_C",
+                "MB_OUTLET_TEMP_C",
+                "PDB_L_TEMP_C",
+                "PDB_R_TEMP_C"
+            ],
+            "Name": "Stepwise_OTHERS",
+            "NegativeHysteresis": 0.0,
+            "Output": [
+                10.0
+            ],
+            "PositiveHysteresis": 0.0,
+            "Reading": [
+                5.0
+            ],
+            "Type": "Stepwise",
+            "Zones": [
+                "Zone 1"
+            ]
+        },
         {
             "Address": "0x4B",
             "Bus": 9,
-- 
2.53.0-Meta

